intake_omnisci/catalog.py

```diff
@@ -0,0 +1,79 @@
+from intake.catalog.base import Catalog
+from intake.catalog.local import LocalCatalogEntry
+import pymapd
+
+from . import __version__
+
+
+class OmniSciCatalog(Catalog):
+    """
+    Makes data sources out of known tables in the given OmniSci database.
+    """
+
+    name = "omnisci_cat"
+    container = "catalog"
+    version = __version__
+
+    def __init__(
+        self,
+        uri=None,
+        user=None,
+        password=None,
+        host=None,
+        port=6274,
+        dbname=None,
+        protocol="binary",
+        metadata=None,
+        **kwargs,
+    ):
+        """
+        Load data from an OmniSci database.
+        Can take *either* a SQLAlchemy-compliant connection string,
+        or a set uf user, password, host, port, dbname.
+
+        Parameters:
+            uri: str
+                a valid OmniSci uri in the form 'mapd://user:password@host:port/database'.
+            user: str
+                The user name.
+            password: str
+                The user password.
+            host: str
+                The hostname for the database server.
+            port: int
+                The port for the database server.
+            dbname: str
+                The name of the database to connect to.
+            protocol: str
+                The protocol for the database server.
+            metadata: dict
+                The metadata for the source.
+        """
+        self._init_args = {
+            "uri": uri,
+            "user": user,
+            "password": password,
```

Comment:
Is it better practice, or particularly necessary to implement base.Catalog when defining Intake Sources?

Reply:
No, it's not necessary. This is a separate intake source which allows you to flip through the individual tables of a database, and construct sources for each of those. I followed the lead of `intake-sql` here.